libs/kg-commons/src/main/java/eu/ebrains/kg/commons/jsonld/NormalizedJsonLd.java

```diff
@@ -115,4 +115,20 @@ else if(invitation){
         return this;
     }
 
+    public void removeAllPropertiesWhenNoPayload() {
```

Comment:
These new methods are not generic enough to be located in "NormalizedJsonLd". They contain EBRAINS specific semantics (the NormalizedJsonLd is meant to be semantic-agnostic). It could either be moved to the CoreInstanceController (similar to "resolveAlternatives" or "enrichWithPermissionInformation"  - or by building a helper class with static methods to manipulate the response object to extract this very specific logic

Reply:
I made a new commit with the changes. Please notice that for deleting payload, I decided to make the methods in InstancesRepository since there is no dependency between this repository and CoreInstanceController.